src/content/en/updates/2019/02/intersectionobserver-v2.md

```diff
@@ -0,0 +1,210 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Augment IntersectionObserver to report information about occlusion and visual effects.
+
+{# wf_updated_on: 2019-02-11 #}
+{# wf_published_on: 2019-02-06 #}
+{# wf_tags: intersectionobserver #}
+{# wf_featured_image: /web/updates/images/generic/timeline.png #}
+{# wf_featured_snippet: Lorem ipsum. #}
+{# wf_blink_components: Blink>Layout #}
+
+# Trust is Good, Observation is Better—Intersection Observer&nbsp;v2 {: .page-title}
+
+{% include "web/_shared/contributors/thomassteiner.html" %}
+
+<div class="clearfix"></div>
+
+Intersection Observer&nbsp;v1 is one of these APIs that's probably universally loved, and, now that
+[Safari supports it](https://webkit.org/blog/8582/intersectionobserver-in-webkit/) as well,
+it's also finally universally usable in all major browsers. For a quick refresher of the API,
+I recommend watching [Surma](../../../resources/contributors/surma)'s
+[Supercharged Microtip](https://www.youtube.com/embed/kW_atFXMG98) on Intersection
+Observer&nbsp;v1—also embedded below for your viewing pleasure—or reading Surma's in-depth
+[article](../../2016/04/intersectionobserver).
+People have used Intersection Observer&nbsp;v1 for a wide range of use cases, like, for example,
+[lazy loading of images and videos](../../../fundamentals/performance/lazy-loading-guidance/images-and-video/),
+[being notified when elements reach `position: sticky`](../../2017/09/sticky-headers),
+[fire analytics events](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/visibility-manager.js),
+and many more.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/kW_atFXMG98" frameborder="0"
+    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
+</iframe>
+
+For the full details, check out the
+[Intersection Observer docs on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API),
+but as a short reminder, this is what the Intersection Observer&nbsp;v1 API looks like in the most
+basic case:
+
+```javascript
+const onIntersection = (entries) => {
+  entries.forEach((entry) => {
+    if (entry.isIntersecting) {
+      console.log(entry);
+    }
+  });
+};
+
+const observer = new IntersectionObserver(onIntersection);
+observer.observe(document.querySelector('#some-target'));
+```
+
+## What's challenging with Intersection Observer&nbsp;v1?
+
+Let's get this straight: there's nothing wrong with Intersection Observer&nbsp;v1, yet there are
```

Comment:
```suggestion
To be clear, Intersection Observer&nbsp;v1 is great but it's not perfect. There are
```

Take-it-or-leave-it suggestion. If there was nothing wrong with IOv1 then we'd have no need for V2. It's a good API but misses features for some really important use cases. Also, "let's get this straight" sounds argumentative; as if someone came into the room bad-mouthing IOv1 and you're going to get in their face to defend your beloved API.

Reply:
Edited, thanks.